crates/rerun_c/src/error.rs

```diff
@@ -0,0 +1,64 @@
+use crate::{CError, CErrorCode};
+
+impl CError {
+    #[allow(unsafe_code)]
+    pub(crate) fn write_error(error: *mut CError, code: CErrorCode, message: &str) {
+        let error = unsafe { error.as_mut() };
+        let Some(error) = error else {
+            return;
+        };
+
+        error.code = code;
+
+        let bytes = message.bytes();
+        let message_byte_length_excluding_null = bytes.len().min(error.message.len() - 1);
```

Comment:
This can produce invalid UTF-8 if dropping some bytes of a multi-byte sequence

Reply:
good catch! I suppose I'll need to do some character based method here instead

Reply:
ooof. for that I need to depend on https://crates.io/crates/unicode-segmentation
we luckily already do in some places in our tree

Reply:
Hm okay no I think chars might be enough. This is about being valid utf8 and not about being valid grapheme clusters that people can necessarily read

Reply:
solved and added a test